config/v1alpha2/types_insights.go

```diff
@@ -0,0 +1,115 @@
+	// spec holds user settable values for configuration
+	// +required
+	Spec InsightsDataGatherSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status InsightsDataGatherStatus `json:"status"`
+}
+
+type InsightsDataGatherSpec struct {
+	// gatherConfig spec attribute includes all the configuration options related to
+	// gathering of the Insights data and its uploading to the ingress.
+	// +optional
+	GatherConfig GatherConfig `json:"gatherConfig,omitempty"`
+}
+
+type InsightsDataGatherStatus struct{}
+
+// gatherConfig provides data gathering configuration options.
+type GatherConfig struct {
+	// dataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data. Valid values are "ClearText" and "ObfuscateNetworking".
+	// When set to ClearText the data is not obfuscated.
+	// When set to ObfuscateNetworking the IP addresses and the cluster domain name are obfuscated.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is ClearText.
+	// +optional
+	DataPolicy DataPolicy `json:"dataPolicy,omitempty"`
+	// gatherers is a list of gatherers configurations.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// Run the following command to get the names of last active gatherers:
+	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
+	// +kubebuilder:validation:MaxItems=100
+	// +listType=atomic
+	// +optional
+	Gatherers []GathererConfig `json:"gatherers,omitempty"`
+}
+
+// state declares valid gatherer state types.
+// +kubebuilder:validation:Enum="";Enabled;Disabled
+type GathererState string
+
+// gathererConfig allows to configure specific gatherers
+type GathererConfig struct {
+	// name is the name of specific gatherer
+	// +kubebuilder:validation:MaxLength=256
+	// +required
+	Name string `json:"name"`
+	// state allows you to configure specific gatherer. Valid values are "Enabled", "Disabled" and omitted.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default.
+	// The current default is Enabled.
+	// +optional
+	State GathererState `json:"state"`
+}
+
+const (
+	// No data obfuscation
+	ClearText DataPolicyOption = "ClearText"
+	// IP addresses and cluster domain name are obfuscated
+	ObfuscateNetworking DataPolicyOption = "ObfuscateNetworking"
+	// Gatherer state marked as disabled, which means that the gatherer will not run.
+	Disabled GathererState = "Disabled"
+	// Gatherer state marked as enabled, which means that the gatherer will run.
+	Enabled GathererState = "Enabled"
+)
+
+// dataPolicyOption declares valid data policy options
+// +kubebuilder:validation:Enum="";ClearText;ObfuscateNetworking
```

Comment:
If this is an enum of values used in a list, why would a user add `""` to the list? Is that valid?

Reply:
I am not 100% why is it here, I have taken this from our `v1alpha1`. But I think we could probably get rid of that if I am not overlooking something WDYT @tremes @ncaak

Reply:
humm I don't know to be honest. I don't think there's a special meaning behind that. Originally it was a string, so maybe it said that an empty string was accepted.